### PR TITLE
Bug SDN-3458: HyperShift: Differentiate resources deployed by different CNO instances in status manager

### PR DESCRIPTION
**Pull request by kyrtapz**
CNO should only reconcile its status based on the resources it rendered.
In the management cluster there is the default CNO and one CNO per hosted clusters control namespace.
By using a label value set to the infrastructure name we can differentiate the resources rendered by different CNO instances.

In stand-alone environments the value is a non-empty constant so it doesn't overlap with older CNO versions.

Additionally start watching deployments in the hosted cluster control plane namespace sine we added CNCC and multus admission controller to it.

### CODE REVIEW

File: pkg/controller/statusmanager/status_manager.go [hunk 1]
```diff
@@ -113,6 +114,16 @@ func New(client cnoclient.Client, name string) *StatusManager {
 		ssInformers:  map[string]cache.SharedIndexInformer{},
 		ssListers:    map[string]StatefulSetLister{},
 	}
+	cluster := ""
+	if status.hyperShiftConfig.Enabled {
+		cluster = status.hyperShiftConfig.Name
```
Comment (zshi-redhat) [inline pkg/controller/statusmanager/status_manager.go:119]: Should this be `bootstrapResult.Infra.InfraName`?
I noticed infraName is assigned to `names.GenerateStatusLabel` label.

Comment (kyrtapz) [reply pkg/controller/statusmanager/status_manager.go:119]: yes, absolutely. 
It is just a leftover from my previous approach

Comment (kyrtapz) [reply pkg/controller/statusmanager/status_manager.go]: Done, the PR is ready for a review @zshi-redhat

File: pkg/controller/statusmanager/pod_watcher.go [hunk 2]
```diff
@@ -75,11 +76,11 @@ func (s *StatusManager) initInformersFor(clusterName, namespace string, stateful
 
 // AddPodWatcher wires up the PodWatcher to the controller-manager.
 func (s *StatusManager) AddPodWatcher(mgr manager.Manager) error {
-	s.initInformersFor("", metav1.NamespaceAll, false)
```
Comment (patjlm) [inline pkg/controller/statusmanager/pod_watcher.go:78]: in the hypershift hosted control plane case, why would we watch all namespaces ?

Comment (kyrtapz) [reply pkg/controller/statusmanager/pod_watcher.go:78]: We want to watch all of the resources created by the CNO in the hosted cluster.

Comment (patjlm) [reply pkg/controller/statusmanager/pod_watcher.go:78]: but that would not be on the same API endpoint, would it? This question is probably out of scope of this PR, sorry. Happy to take it elsewhere.

Comment (kyrtapz) [reply pkg/controller/statusmanager/pod_watcher.go:78]: Correct, CNO talks to both the management and hosted clusters API servers. 
`initInformersFor` uses the `clusterName` to determine which client to use (empty string means hosted cluster API server)

**Comment by kyrtapz** [comment]
test e2e-hypershift-ovn

**Pull request merged**